metadata.yaml

```diff
@@ -27,9 +27,8 @@ containers:
 resources:
   alertmanager-image:
     type: oci-image
-    # The officially supported image is ubuntu/prometheus-alertmanager
     description: OCI image for alertmanager
-
+    upstream-source: ubuntu/prometheus-alertmanager
```

Comment:
This is a deprecated field aiui

Reply:
It's not a deprecated field AFAIK. We use it extensively in the Kubeflow bundle. Juju itself won't do anything with the field, but it a) acts as a comment about where someone should look if they want to inspect the Dockerfile for the charm they're running, and b) is also used in a lot of tooling around charm deployment and publishing.

See here for how we use that field when publishing a charm: https://github.com/canonical/kubeflow-dashboard-operator/blob/798e432c/.github/workflows/publish.yaml#L33

When Launchpad starts building charms for charmhub.io, we'll want some sort of similar functionality, and so may want to just use this existing field.

Reply:
Also, this field should probably have a version specifier such as `0.21-20.04_beta`, or at least a sha hash if we want to use the current latest image

Reply:
Yeh, I guess for charms published to old charm store it makes sense, though I suspect the O11y team will never publish there. Interesting point about encapsulating the information in the field to make the automation tooling easier for publishing though, that is a nice point.

Even though the workflow is to upload an oci-resource, then associate that with a release, I could see why having this in metadata.yaml would make it easier to automate.

Reply:
Yes, `upstream-source` is the only consistent machine-readable place to document the image, which is also useful for local automation.

Reply:
> this field should probably have a version specifier such as `0.21-20.04_beta`

Good point! We haven't been freezing versions at all so far. One difficulty is remembering to match it with the (opaque) uploaded image resource on charmhub.

Reply:
What is the reason for juju not reading this? It would be really nice to not have to specify --resource every time I deploy a local charm.

Reply:
@dstathis for now you'd have to use yq, for example: `yq eval '.resources[].upstream-source' metadata.yaml`, assuming you only have one image.